src/AsmResolver/Shims/ArrayShim.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+
+namespace AsmResolver.Shims;
+
+/// <summary>
+/// Provides compatibility shims for the <see cref="Array"/> class for builds that target older .NET framework versions.
+/// </summary>
+public static class ArrayShim
+{
+    /// <summary>
+    /// Obtains a singleton instance of an empty array of the provided type.
+    /// </summary>
+    /// <typeparam name="T">The type to get the empty array for.</typeparam>
+    /// <returns>The empty array.</returns>
+#if !NET35
+    public static T[] Empty<T>() => Array.Empty<T>();
+#else
+    public static T[] Empty<T>() => ArrayHelper<T>.Empty;
+
+    private static class ArrayHelper<T>
+    {
+        public static readonly T[] Empty = new T[0];
+    }
+#endif
```

Comment:
Collection expressions might be supported on NET35. If so, I think you could use `[]` everywhere instead of having a shim.

Reply:
I think that when `Array.Empty<T>()` is not available, `[]` will always construct a new array. Probably worth double checking though.

Reply:
This is indeed the case. On .NET 3.5 the following C#
```csharp
public static int[] ReturnEmpty() => [];
```
results in the following CIL, creating a new empty array on every call.
```
.method public static hidebysig int32[] ReturnEmpty() cil managed
{
    .maxstack 8

    IL_0000: ldc.i4.0
    IL_0001: newarr valuetype [mscorlib] System.Int32
    IL_0006: ret
}
```